src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/glide/MauiCustomViewTarget.java

```diff
@@ -17,25 +17,32 @@
 public class MauiCustomViewTarget extends CustomViewTarget<ImageView, Drawable> {
     private final ImageLoaderCallback callback;
     private boolean completed;
+    private boolean cleared;
 
     public MauiCustomViewTarget(@NonNull ImageView view, ImageLoaderCallback callback) {
         super(view);
 
         this.completed = false;
+        this.cleared = false;
         this.callback = callback;
     }
 
     @Override
     protected void onResourceCleared(@Nullable Drawable placeholder) {
-		this.view.setImageDrawable(placeholder);
+        this.cleared = true;
     }
 
     @Override
     public void onLoadFailed(@Nullable Drawable errorDrawable) {
         if (completed)
             return;
+
         this.completed = true;
 
+        if (this.cleared) {
+            this.view.setImageDrawable(null);
+        }
```

Comment:
Should something set `cleared` back to `false` at some point after it has become `true`?

Reply:
I've followed the same approach as `this.completed` because this entire class is basically like a `Task`.. once it is completed (failed in this case) nothing can bring it back.
The instance will not be reused by the caller.